crates/e2e/tests/e2e/flashloans.rs

```diff
@@ -0,0 +1,280 @@
+    let [trader] = onchain.make_accounts(to_wei(1)).await;
+    let trader = Safe::deploy(trader.clone(), &web3).await;
+
+    // AAVE token tracking how much USDC is deposited by a user
+    let ausdc = addr!("98c23e9d8f34fefb1b7bd6a91b7ff122f4e16f5c");
+    let weth = &onchain.contracts().weth;
+    let settlement = &onchain.contracts().gp_settlement;
+
+    // transfer some USDC from a whale to our trader
+    let usdc = ERC20::at(&web3, addr!("a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"));
+    let usdc_whale_mainnet = addr!("28c6c06298d514db089934071355e5743bf21d60");
+    let usdc_whale = forked_node_api
+        .impersonate(&usdc_whale_mainnet)
+        .await
+        .unwrap();
+
+    // transfer $50K + 1 atom to the safe (50K for the test and 1 atom for passing
+    // the minimum balance test before placing the order)
+    let collateral_amount = to_wei_with_exp(50_000, 6);
+    tx!(
+        usdc_whale,
+        usdc.transfer(trader.address(), collateral_amount + U256::from(1))
+    );
+    // approve vault relayer to take safe's sell tokens
+    trader
+        .exec_call(usdc.approve(onchain.contracts().allowance, collateral_amount))
+        .await;
+
+    // Approve AAVE to take the collateral
+    let aave_pool = IAavePool::at(&web3, addr!("87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2"));
+    trader
+        .exec_call(usdc.approve(aave_pool.address(), collateral_amount))
+        .await;
+
+    // Deposit collateral
+    trader
+        .exec_call(aave_pool.supply(
+            usdc.address(),    // token
+            collateral_amount, // amount
+            trader.address(),  // on_behalf
+            0,                 // referral code
+        ))
+        .await;
+    assert!(balance(&web3, trader.address(), ausdc).await >= collateral_amount);
+
+    tracing::info!("wait a bit to make `borrow()` call work");
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    // Borrow 1 WETH against the collateral
+    let flashloan_amount = to_wei(1);
+    trader
+        .exec_call(aave_pool.borrow(
+            onchain.contracts().weth.address(), // borrowed token
+            flashloan_amount,                   // borrowed amount
+            2.into(),                           // variable interest rate mode
+            0,                                  // referral code
+            trader.address(),                   // on_behalf
+        ))
+        .await;
+
+    // allow aave pool to take back borrowed WETH on `repay()`
+    // could be replaced with `permit` pre-hook or `repayWithPermit()` for
+    // borrowed tokens that support `permit`
+    trader
+        .exec_call(
+            onchain
+                .contracts()
+                .weth
+                .approve(aave_pool.address(), to_wei(1)),
+        )
+        .await;
+
+    let current_safe_nonce = trader.nonce().await;
+
+    // Build appdata that does:
+    // 1. take out a 1 WETH flashloan for the trader
```

Comment:
1. is not actually part of the appdata right, it's already done above?

Reply:
Did you confuse flashloan with borrowing the WETH (which indeed happens above)?

The flashloan is not explicitly encoded in an interaction but it's still part of the appdata as the `flashloan` hint. Updated the comment to make this clearer.